engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbFormState.java

```diff
@@ -137,8 +137,9 @@ public Optional<PersistedForm> findLatestFormById(
       return Optional.empty();
     }
 
-    formByTenantAndIdCache.put(new TenantIdAndFormId(tenantId, formId), persistedForm);
-    return Optional.of(persistedForm).map(PersistedForm::copy);
+    final PersistedForm copiedForm = persistedForm.copy();
+    formByTenantAndIdCache.put(new TenantIdAndFormId(tenantId, formId), copiedForm);
+    return Optional.of(copiedForm);
```

Comment:
out of interest, so the underlying cause is that
```
final PersistedForm persistedForm =
        formByIdAndVersionColumnFamily.get(tenantAwareIdAndVersionKey);
```
or generally `TransactionalColumnFamily.get` by design returns a mutable object, right?

Reply:
Yes, it fills the `dbPersistentForm` singular object existing in the DbFormState class with retrieved data and returns that object back.

Reply:
That seems like an easy and non obvious mistake to make, I wonder why it was designed that way. Will follow-up to learn or at least ensure we have the needed awareness of this.

Reply:
Yeah, we discussed the same thing with @korthout. It is designed to be performant and prevents Java object creation. Instead, it uses a single object and re-fill it on change.

Also, I think, there is a design improvement for the `get` method where it is a method with side effect but returns an object back. To emphasise its side effect, it should return `void` (probably the name should change from `get` as well). Therefore, db api users will know some state is mutated.

Reply:
@megglos All column families are provided with objects representing the key and value. These are used by operations of the column family to perform their operations and to fill with output. Rather than providing constructors as suppliers, the object is reused for performance. In our code, we always need to copy objects before storing them because the object is likely reused in successive processing.